File: 広告/アドフリくん/adfurikunSDK_Android_moviereward_3.20.0/SampleJava/app/src/main/java/sample/adfurikunsdk/jp/samplejava/SimpleNativeAd.java
package sample.adfurikunsdk.jp.samplejava;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.TextView;

import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunMovieError;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunNativeAd;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunNativeAdInfo;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunNativeAdLoadListener;
import jp.tjkapp.adfurikunsdk.moviereward.AdfurikunNativeAdVideoListener;

/**
 * ネイティブ広告用サンプル
 */
public class SimpleNativeAd extends Activity {

    private TextView mTitle;
    private TextView mDescription;
    private FrameLayout mNativeAdViewFrame;
    private Button mLoad;
    private Button mPlay;

    /**
     * ネイティブ広告インスタンス
     */
    private AdfurikunNativeAd mNativeAd;
    /**
     * ネイティブ広告情報
     */
    private AdfurikunNativeAdInfo mNativeAdInfo;

    /**
     * 広告読み込みリスナー
     */
    private AdfurikunNativeAdLoadListener mLoadListener = new AdfurikunNativeAdLoadListener() {
        @Override
        public void onNativeAdLoadFinish(AdfurikunNativeAdInfo adInfo, String appId) {
            // 広告の準備完了直後に実行する処理を記述してください
            Utils.log("onNativeAdLoadFinish appId = " + appId);
            if (adInfo != null) {
                mNativeAdInfo = adInfo;
                String title = adInfo.getTitle();
                String description = adInfo.getDescription();
                mTitle.setText(title);
                mDescription.setText(description);
                mPlay.setEnabled(true);
            }
        }

        @Override
        public void onNativeAdLoadError(AdfurikunMovieError error, String appId) {
            // 広告の準備失敗直後に実行する処理を記述してください
            String errorType = "";
            if (error != null) {
                errorType = error.getErrorType().name();
            }
            Utils.log("onNativeAdLoadError appId = " + appId + ", errorType = " + errorType);
        }
    };

    /**
     * 広告再生リスナー
     */
    private AdfurikunNativeAdVideoListener mVideoListener = new AdfurikunNativeAdVideoListener() {
        @Override
        public void onNativeAdViewPlayStart(String appId) {
            // 広告の再生開始直後に実行する処理を記述してください
            Utils.log("onNativeAdViewPlayStart appId = " + appId);
        }

        @Override
        public void onNativeAdViewPlayFinish(String appId, boolean isVideoAd) {
            // 広告の再生が完了したら実行する処理を記述してください
            Utils.log("onNativeAdViewPlayFinish appId = " + appId + ", isVideoAd = " + isVideoAd);
        }

        @Override
        public void onNativeAdViewPlayFail(String appId, AdfurikunMovieError error) {
            // 広告の再生に失敗した時に実行する処理を記述してください
            String errorType = "";
            if (error != null) {
                errorType = error.getErrorType().name();
            }
            Utils.log("onNativeAdViewPlayFail appId = " + appId + ", errorType = " + errorType);
        }

        @Override
        public void onNativeAdViewClicked(String appId) {
            // 広告の画面をクリックした時に実行する処理を記述してください
            Utils.log("onNativeAdViewClicked appId = " + appId);
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_native_ad);
        setTitle(getIntent().getStringExtra("title"));

        mDescription = findViewById(R.id.description);
        mTitle = findViewById(R.id.title);
        mNativeAdViewFrame = findViewById(R.id.native_ad_view_frame);
        mLoad = findViewById(R.id.load);
        mPlay = findViewById(R.id.play);

        // ボタンを押すと広告を読み込む
        mLoad.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mNativeAd != null) {
                    mNativeAd.load();
                }
            }
        });

        // ボタンを押すと広告を表示する
        mPlay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mNativeAd != null) {
                    mNativeAd.play();
                }
            }
        });

        // ネイティブ広告のインスタンスを生成する
        // 広告枠IDとActivityを引数で渡してください
        mNativeAd = new AdfurikunNativeAd(this, AdfurikunConst.ADFURIKUN_NATIVE_AD_APPID,
                Utils.convertDpToPx(this, 320),
                Utils.convertDpToPx(this, 180));
        mNativeAd.setAdfurikunNativeAdLoadListener(mLoadListener);
        mNativeAd.setAdfurikunNativeAdVideoListener(mVideoListener);

        mNativeAdViewFrame.addView(mNativeAd.getNativeAdView());
    }

    @Override
    protected void onDestroy() {
        if (mNativeAd != null) {
            mNativeAd.onDestroy();
        }
        super.onDestroy();
    }
}

File: 育成中コンテンツ/広告/アドフリくん/adfurikunSDK_Android_moviereward_3.20.0/SampleKotlin/app/src/main/java/jp/glossom/sample/samplekotlin/SimpleNativeAd.kt
package jp.glossom.sample.samplekotlin

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.FrameLayout
import android.widget.TextView
import jp.tjkapp.adfurikunsdk.moviereward.*

/**
 * ネイティブ広告用サンプル
 */
class SimpleNativeAd : Activity() {

    private lateinit var mTitle: TextView
    private lateinit var mDescription: TextView
    private lateinit var mNativeAdViewFrame: FrameLayout
    private lateinit var mLoad: Button
    private lateinit var mPlay: Button

    /** ネイティブ広告インスタンス */
    private var mNativeAd: AdfurikunNativeAd? = null
    /** ネイティブ広告情報 */
    private var mNativeAdInfo: AdfurikunNativeAdInfo? = null

    /** 広告読み込みリスナー */
    private val mLoadListener = object : AdfurikunNativeAdLoadListener {
        override fun onNativeAdLoadFinish(adInfo: AdfurikunNativeAdInfo?, appId: String?) {
            // 広告の準備完了直後に実行する処理を記述してください
            Utils.log("onNativeAdLoadFinish appId = $appId")
            adInfo?.let {
                mNativeAdInfo = it
                mTitle.text = it.title
                mDescription.text = it.description
                mPlay.isEnabled = true
            }
        }

        override fun onNativeAdLoadError(error: AdfurikunMovieError?, appId: String?) {
            // 広告の準備失敗直後に実行する処理を記述してください
            Utils.log("onNativeAdLoadError appId = $appId, errorType = ${error?.errorType?.name}")
        }
    }

    /** 広告再生リスナー */
    private val mVideoListener = object : AdfurikunNativeAdVideoListener {
        override fun onNativeAdViewPlayStart(appId: String?) {
            // 広告の再生開始直後に実行する処理を記述してください
            Utils.log("onNativeAdViewPlayStart appId = $appId")
            mPlay.isEnabled = false
        }

        override fun onNativeAdViewPlayFinish(appId: String?, isVideoAd: Boolean) {
            // 広告の再生に失敗した時に実行する処理を記述してください
            Utils.log("onNativeAdViewPlayFinish appId = $appId, isVideoAd = $isVideoAd")
        }

        override fun onNativeAdViewPlayFail(appId: String?, error: AdfurikunMovieError?) {
            // 広告の再生が完了したら実行する処理を記述してください
            Utils.log("onNativeAdViewPlayFail appId = $appId, errorType = ${error?.errorType?.name}")
        }

        override fun onNativeAdViewClicked(appId: String?) {
            // 広告の画面をクリックした時に実行する処理を記述してください
            Utils.log("onNativeAdViewClicked appId = $appId")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_native_ad)
        title = intent.getStringExtra("title")

        mTitle = findViewById(R.id.title)
        mDescription = findViewById(R.id.description)
        mNativeAdViewFrame = findViewById(R.id.native_ad_view_frame)
        mLoad = findViewById(R.id.load)
        mPlay = findViewById(R.id.play)

        // ボタンを押すと広告を読み込む
        mLoad.setOnClickListener {
            mNativeAd?.load()
        }

        // ボタンを押すと広告を表示する
        mPlay.setOnClickListener {
            mNativeAd?.play()
        }

        // ネイティブ広告のインスタンスを生成する
        // 広告枠IDとActivityを引数で渡してください
        mNativeAd = AdfurikunNativeAd(
            this, AdfurikunConst.ADFURIKUN_NATIVE_AD_APPID,
            Utils.convertDpToPx(this, 320),
            Utils.convertDpToPx(this, 180)
        )
        mNativeAd?.setAdfurikunNativeAdLoadListener(mLoadListener)
        mNativeAd?.setAdfurikunNativeAdVideoListener(mVideoListener)

        mNativeAdViewFrame.addView(mNativeAd?.nativeAdView)
    }

    override fun onDestroy() {
        mNativeAd?.onDestroy()
        super.onDestroy()
    }
}
